src/arithmetic/algebraic_expression.h

```diff
@@ -30,7 +30,11 @@ typedef struct AlgebraicExpressionNode AlgebraicExpressionNode;
 
 struct AlgebraicExpressionNode {
     union {
-        GrB_Matrix operand;
+        struct {
+            bool entity_is_node;
+            void *entity;
+            GrB_Matrix mat;
```

Comment:
Suggestion, what if for AlgebraicExpressions we wrap the matrix object in a struct
which will contain src,dest nodes and edge?

Reply:
Hm, this seems like a good idea to me! The AlgebraicExpression wouldn't even take up more space, since the union is 32 bytes.

With this change, we'd traverse the tree to find the operation's src and dest in NewCondTraverseOp?

Reply:
no need to traverse, (I think)
When constructing the expression tree, set the root node wrapped matrix object source and destination node.